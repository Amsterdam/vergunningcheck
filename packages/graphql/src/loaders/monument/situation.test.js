const uuid = require("uuid");
const situation = require("rewire")("./situation.js");
const loader = situation.__get__("loader");

const load = jest.fn(
  (id) => new Promise((resolve) => resolve(`data for ${id}`))
);

// Setup mock
situation.__set__(
  "loader",
  Object.assign({}, loader, {
    load,
  })
);

describe("Situation loader", () => {
  beforeEach(async () => {
    jest.clearAllMocks();
  });
  test("load function", async () => {
    const key = uuid();
    const result = await Promise.all(await situation.load([key]));
    expect(result).toStrictEqual([`data for ${key}`]);
  });

  test("cache hit", async () => {
    const key = uuid();
    const a = await Promise.all(await situation.load([key]));
    const b = await Promise.all(await situation.load([key]));
    expect(a).toStrictEqual(b);
    expect(load.mock.calls.length).toBe(1);
  });

  test("cache miss", async () => {
    const a = await Promise.all(await situation.load([uuid()]));
    const b = await Promise.all(await situation.load([uuid()]));
    expect(load.mock.calls.length).toBe(2);
    expect(a).not.toStrictEqual(b);
  });
});
