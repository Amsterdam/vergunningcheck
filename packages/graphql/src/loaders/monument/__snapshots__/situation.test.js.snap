// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Situation loader Loader internals module result 1`] = `
Array [
  Array [
    Object {
      "monumentId": "c70dd3b7-50e9-43a1-bc87-80bf5bc3906a",
    },
  ],
]
`;

exports[`Situation loader Loader internals module result 2`] = `
Array [
  Array [
    Object {
      "monumentId": "c19e5938-24de-4d41-a6d3-e07d44878246",
    },
  ],
]
`;
